org.eclipse.scanning.api/src/org/eclipse/scanning/api/IValidator.java

```diff
@@ -47,4 +47,18 @@ default void validate(T model) throws ValidationException, InstantiationExceptio
 	default void setService(IValidatorService vservice) {
 		
 	}
+	
+	/**
+	 * Same as Validaiton method above but returns any results sent back by validation.
+	 * 
+	 * @param model
+	 * @return
+	 * @throws ValidationException
+	 * @throws InstantiationException
+	 * @throws IllegalAccessException
+	 */
+	default Object validateWithReturn(T model) throws ValidationException, InstantiationException, IllegalAccessException {

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Refactor this method to throw at most one checked exception instead of: java.lang.InstantiationException, java.lang.IllegalAccessException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1160)